### DISCUSSION TITLE
Progressive Data Display from a Single Provider in Flutter

### DISCUSSION THREAD

**Discussion by MichaelLucca** [rrousselGit/riverpod#3357] [Q&A] [answered]
**Description**
Currently, I'm using a single provider in Flutter to fetch data from multiple API calls within my application. However, I'm encountering difficulty in managing the progressive display of data in my UI. I want to display portions of the UI as data is received, rather than having to wait for all API calls to complete.

I've implemented a solution to achieve progressive data display from a single API provider in Flutter, and I'd like to gather feedback to ensure it's the recommended approach.

**The Provider**
![Screenshot 2024-02-22 alle 16 53 16](https://github.com/rrousselGit/riverpod/assets/89570547/27254432-3bf0-4c69-9bf9-7a36937570b3)

**The UI**
![Screenshot 2024-02-22 alle 16 54 10](https://github.com/rrousselGit/riverpod/assets/89570547/4a813614-c4b0-4d94-8222-d3e248099dc4)

**Comment by rrousselGit** [comment, marked as answer]
Don't have the provider read itself using `ref.read`. I'm surprised this even works honestly.

Rather than:

```dart
ref.read(storeItemDetailsProvider(...).notifier).state = ...;
```
do:
```dart
state = ...;
```

**Reply by MichaelLucca** [reply]
Thanks a lot @rrousselGit 🤙🏻
